src/utils.ts

```diff
@@ -60,3 +60,21 @@ export function getSplicePosition(
   }
   return splicePosition;
 }
+
+export function toArrayBuffer(
+  arr: Uint8Array | ArrayBuffer | DataView | object
+): ArrayBufferLike {
+  if (arr instanceof Uint8Array) {
+    return arr.buffer.slice(arr.byteOffset, arr.byteOffset + arr.byteLength);
```

Comment:
I found this bug while updating the above tests.  Your original implementation was passing the buffer directly to the caller, and I _think_ that was a bug because that buffer will almost certainly be bigger than your incoming content.  Here, we're instead returning a new slice that contains only the bytes our incoming `Uint8Array` that we need.